drafts/two-way-binding-signals-and-query-params.md

```diff
@@ -0,0 +1,754 @@
+
+If you want to learn more about what signals are and a few interesting historical details about them, reach out to our amazing previous article [Signals, Signals everywhere][signalsEverywhereArticle] by [Lucas Schnüriger][lucasProfile].
+
+The whole concept of signals is a pretty new topic in the Angular community. In fact, just a few weeks ago, [Angular 17.2][angular17.2] was released, introducing the possibility of using two-way binding along with signals. And not long before, [Angular 17.1][angular17.1] released a new API to specify `@Input()` while working with results as signals. 
+
+Wait a minute, really? Weren't those important APIs already available when the Angular team released the first stable version of signals? Isn't two-way binding or, even more, parent-to-child communication basic features of the framework? Yes, we could claim that but the Angular team is taking a gradual strategy to introduce all these new concepts into the framework. And that's probably a good thing, since they are learning and listening from the community during the process. You can also influence and give your opinion in the [open discussions][angularDiscussions].
+
+In today's article, we'll experiment and push to the limit the current state of signals in the Angular framework. For that, we will implement an exciting Two-way binding mechanism to connect signals with Angular's router state and the browser url. Are you ready? 
+
+## The Challenge
+
+Let's start with a simple application:
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  template: ` <h1>Signals Demo</h1> `,
+})
+export class AppComponent {}
+```
+
+![Initial application](https://cdn.hashnode.com/res/hashnode/image/upload/v1710536390017/q5rnF-2Oc.png?auto=format&w=500)
+
+This is pretty much what you get when you generate a new Angular application nowadays. But without all the boilerplate and some basic styles.
+
+Let's start by consuming query params in the traditional way: using `RxJS`:
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  imports: [AsyncPipe],
+  template: `
+    <h1>Signals Demo</h1>
+    <p>Your first name is: {{ firstName$ | async }}</p>
+  `,
+})
+export class AppComponent {
+  private activatedRoute = inject(ActivatedRoute);
+
+  firstName$ = this.activatedRoute.queryParams.pipe(
+    map((allQueryParams) => allQueryParams["firstName"])
+  );
+}
+```
+
+![Consuming query params with RxJS](https://cdn.hashnode.com/res/hashnode/image/upload/v1710536464769/FbnjIrApA.gif?auto=format)
+
+Great! this works and it's not necessarily inconvenient since Angular is handling the tedious task of managing subscriptions when we use the `async` pipe. Now let's see how this would look like with signals:
```

Comment:
```suggestion
Great! This works and it's not necessarily inconvenient since Angular is handling the tedious task of managing subscriptions when we use the `async` pipe. Now let's see how this would look like with signals.
```